deployer/deployer.go

```diff
@@ -15,101 +16,124 @@ import (
 	"github.com/pkg/errors"
 	client "github.com/threefoldtech/grid3-go/node"
 	"github.com/threefoldtech/grid3-go/subi"
-	proxy "github.com/threefoldtech/grid_proxy_server/pkg/client"
+	"github.com/threefoldtech/grid3-go/workloads"
 	"github.com/threefoldtech/substrate-client"
 	"github.com/threefoldtech/zos/pkg/gridtypes"
 )
 
-// Deployer interface to be used for any deployer
-type Deployer interface {
-	Deploy(ctx context.Context, sub subi.SubstrateExt, oldDeployments map[uint32]uint64, newDeployments map[uint32]gridtypes.Deployment) (map[uint32]uint64, error)
-	GetDeployments(ctx context.Context, sub subi.SubstrateExt, dls map[uint32]uint64) (map[uint32]gridtypes.Deployment, error)
+// DeployerInterface to be used for any deployer
+type DeployerInterface interface {
+	Deploy(ctx context.Context,
+		oldDeploymentIDs map[uint32]uint64,
+		newDeployments map[uint32]gridtypes.Deployment,
+		newDeploymentsData map[uint32]workloads.DeploymentData,
+		newDeploymentSolutionProvider map[uint32]*uint64,
+	) (map[uint32]uint64, error)
+
+	Cancel(ctx context.Context,
+		oldDeploymentIDs map[uint32]uint64,
+		newDeployments map[uint32]gridtypes.Deployment,
+	) (map[uint32]uint64, error)
+
+	GetDeployments(ctx context.Context, dls map[uint32]uint64) (map[uint32]gridtypes.Deployment, error)
+
+	Wait(
```

Comment:
`Wait` is only used internally in the `Deployer` struct is it needed in the `DeployerInterface`?